testing/scenario_app/ios/Scenarios/ScenariosTests/AppLifecycleTests.m

```diff
@@ -140,5 +140,13 @@ - (void)testLifecycleChannel {
   XCTAssertEqualObjects(
       lifecycleEvents, expectedStates,
       @"AppLifecycleState transitions while presenting a second time not as expected");
+
+  // Dismantle.
+  [engine.lifecycleChannel setMessageHandler:nil];
+  [flutterVC dismissViewControllerAnimated:NO completion:nil];
+  flutterVC = nil;
```

Comment:
flutterVC and rootVC shouldn't need to be set to nil, it should dealloc after this test returns?

What exactly was the problem here?  Was state bleeding between tests?

Reply:
ya, setting them to nil was just voodooism. Removed in the next PR. 
The real meat was getting rid of the assert closure on top which would have failed after the test ended if more things happened with the vc as it's dismissed. The VC is also held onto by the engine unless removed explicitly.